### PR TITLE
The great package update of late may®

### PR DESCRIPTION
**Pull request by willsb**
Package consolidation. This removes some warnings. I’ve checked all the packages before updating, so all should be safe.

## Important Dependencies:
- System.Reactive.Linq (`3.0.0` || `3.1.1`) -> `4.0.0`   
  - This is a major release mainly because it deprecates old stuff (RIP WINDOWS PHONE), not because of huge breaking changes
- Realm `2.1.0` -> `3.0.0`
  - This is a major release because of the updates to the Realm backend platform, which we don’t use/care about. The .net assembly itself hasn’t changed much
- Microsoft.AppCenter `1.3.0` -> `1.6.1`
- Microsoft.AppCenter.Analytics `1.3.0` -> `1.6.1`
- Microsoft.AppCenter.Crashes `1.3.0` -> `1.6.1`
  - I read the changelogs and there are no breaking changes in those updates, only bug fixes
- Xamarin.Firebase.iOS `4.0.4` -> `4.0.5.1`
- Xamarin.Firebase.iOS.Analytics `4.0.4` -> `4.0.5.1`
- Xamarin.Firebase.iOS.Core `4.0.8` -> `4.0.13.1`
- Xamarin.Firebase.iOS.CrashReporting `2.0.0` -> `2.0.0.5`
- Xamarin.Firebase.iOS.instanceID `4.0.4` -> `4.0.8`
- Xamarin.Google.iOS.Core `4.1.0` -> `4.1.1.1`
- Xamarin.Google.iOS.SignIn `1.3.0` -> `1.6.1`
- Xamarin.TestCloud.Agent `0.21.1` -> `0.21.5`
  - Again, just lots of small bug fix updates

## Netstandard dependencies (stuff we need because netstandard adds it for a variety of reasons)
- System.Console `4.3.0` -> `4.3.1`
- System.Diagnostics.Contracts `4.0.1` -> `4.3.0`
- System.Dynamic.Runtime `4.0.11` -> `4.3.0`
- System.Net.Http `4.3.2` -> `4.3.3`
- System.Reflection.TypeExtensions `4.3.0` -> `4.5.0`
- System.Runtime.CompilerServices.Unsafe `4.4.0` -> `4.5.0`
- System.Runtime.InteropServices.WindowsRuntime `4.0.1` -> `4.3.0`
- System.Security.Cryptography.Algorithms `4.3.0` -> `4.3.1`
- System.Security.Cryptography.X509Certificates `4.3.0` -> `4.3.2`
- System.XML.ReaderWriter `4.3.0` -> `4.3.1`
- Xamarin.Build.Download `0.4.6` -> `0.4.11`

## Test dependencies:
- FluentAssertions `4.19.4` -> `5.3.2`
- FsCheck.XUnit `2.10.4` -> `2.10.10`
- Microsoft.NET.Test.Sdk  `15.5.0` -> `15.7.2`
- Microsoft.Reactive.Testing -> `3.1.1` -> `4.0.0`

## Reviewing this PR

`Xamarin.Android.*` packages should be updated in a different pull request, since they require more attention.

Use the apps, run UI tests, read the changelogs for the libraries (or check their git issues to see if there's anything that I let slip through the cracks) and just check the diff to see if I missed something.

### CODE REVIEW

**Comment by juanlaube** [comment]
I had problems to build this branch. From a google search, turns out I had to add `--registrar:static` as an additional _mtouch_ argument in the iOS build settings...

**Comment by juanlaube** [comment]
I wrapped that line with `InvokeOnMainThread` to continue testing. Other than that, everything was 👌

**Comment by amulware** [comment]
Can someone check that updating from a previous version to this one works fine (i.e. login and data are still there correctly) - just to make sure the major realm update migrates itself without issues.

**Comment by juanlaube** [comment]
@amulware data is still there and loads correctly :)

File: Toggl.Daneel/Extensions/OnboardingExtensions.cs [hunk 1]
```diff
@@ -133,7 +133,7 @@ public static UIPanGestureRecognizer DismissBySwiping(this DismissableOnboarding
                     visibilityDisposable?.Dispose();
                 });
 
-            cell.AddGestureRecognizer(panGestureRecognizer);
+            cell.InvokeOnMainThread(() => cell.AddGestureRecognizer(panGestureRecognizer));
```
Comment (juanlaube) [inline Toggl.Daneel/Extensions/OnboardingExtensions.cs]: same for line 132: `cel..RemoveGestureRecognizer...`
(I can't add the comment on that line 😒)

**Pull request merged**